packages/common/src/utils/slugs.ts

```diff
@@ -1,8 +1,17 @@
 const sanitizeSlug = (slug: string) =>
     slug
         .toLowerCase()
-        .replace(/[^a-z0-9]+/g, '-') // Replace all non-alphanumeric characters with hyphens
-        .replace(/-+/g, '-') // Replace multiple hyphens with a single hyphen
-        .replace(/^-+|-+$/g, ''); // Trim leading and trailing hyphens
+        // Then replace remaining non-alphanumeric characters with hyphens
+        .replace(/[^a-z0-9]+/g, '-')
+        // Replace multiple hyphens with a single hyphen
+        .replace(/-+/g, '-')
+        // Trim leading and trailing hyphens
+        .replace(/^-+|-+$/g, '');
 
-export const generateSlug = (name: string) => `${sanitizeSlug(name)}`;
+export const generateSlug = (name: string) => {
+    const sanitizedSlug = sanitizeSlug(name);
+    if (sanitizedSlug.length === 0) {
+        return Math.random().toString(36).substring(2, 7);
```

Comment:
I somewhat agree with GitHub CodeQL. We use NanoID in the backend - and I suggest using the same here.